### ISSUE TITLE
Spotifyd cli working / systemd not working

### ISSUE THREAD

**Issue by rwjack** [Spotifyd/spotifyd#1124]
**Description**

```
pulseaudio --start
bluetoothctl
[bluetooth]# connect x
Attempting to connect to x
[CHG] Device x Connected: yes
Connection successful
quit
aplay ~/piano2.wav # works
spotifyd --no-daemon > log.txt
```
^ WORKS AS INTENDED ^
When I select the rpi as source on my desktop spotify, music plays on it

---

```
pulseuadio --start
bluetoothctl
[bluetooth]# connect x
Attempting to connect to x
[CHG] Device x Connected: yes
Connection successful
quit
aplay ~/piano2.wav # works
systemctl start spotifyd
```
^ SPOTIFYD CRASH ^
When I select the rpi as source on my desktop spotify, connection is established, but as soon as I play/pause or change volume, spotifyd crashes.

---

```
systemctl start pulseaudio
bluetoothctl
[bluetooth]# connect x
Attempting to connect to x
Failed to connect: org.bluez.Error.Failed
quit
aplay ~/piano2.wav # doesn't work obviously
spotifyd --no-daemon > log.txt
```
^ NO AUDIO ^
When I select the rpi as source on my desktop spotify, connection is established, but there is no audio output on the rpi, I can switch tracks and change volume though nothing happens.

---

```
systemctl start pulseaudio
bluetoothctl
[bluetooth]# connect x
Attempting to connect to x
Failed to connect: org.bluez.Error.Failed
quit
aplay ~/piano2.wav # doesn't work obviously
systemctl start spotifyd
```
^ NO AUDIO ^
When I select the rpi as source on my desktop spotify, connection is established, but there is no audio output on the rpi. I can switch tracks and change volume though nothing happens.

---

-
`/etc/systemd/system/spotifyd.service`
```
[Unit]
Description=A spotify playing daemon
Documentation=https://github.com/Spotifyd/spotifyd
Wants=sound.target
After=sound.target
Wants=network-online.target
After=network-online.target

[Service]
ExecStart=/usr/local/bin/spotifyd --no-daemon
Restart=always
RestartSec=12

[Install]
WantedBy=default.target
```

-
`/root/.config/spotifyd/spotifyd.conf`:
```
[global]
username = "x"
password = "x"
backend = "alsa"
device = "default"
mixer = "Master"
volume-controller = "alsa" # or alsa_linear, or softvol
#onevent = command_run_on_playback_event
device_name = "devpi"
bitrate = 160 #96|160|320
cache_path = "cache_directory"
volume-normalisation = true
normalisation-pregain = -10
```

-
`/etc/systemd/system/pulseaudio.service`:
```
[Unit]
Description=PulseAudio system server
# DO NOT ADD ConditionUser=!root

[Service]
Type=notify
ExecStart=pulseaudio --daemonize=no --system --realtime --log-target=journal
#ExecStart=pulseaudio --system --disallow-exit --disable-shm --exit-idle-time=-1 --log-target=journal # also tried this
Restart=on-failure

[Install]
WantedBy=multi-user.target
```

---

**Expected behavior**

Expecting music to play when spotifyd is ran as systemd unit. I'm guessing this might be something related to pulseaudio, since I also want pulseaudio to be ran as a systemd unit, insteadof having to log in and start the commands everytime the rpi reboots.

**Logs**
<details><summary>Click to show logs (same for all 4 scenarios)</summary>

```
Loading config from "/root/.config/spotifyd/spotifyd.conf"
No proxy specified
Using software volume controller.
Connecting to AP "ap.spotify.com:443"
Authenticated as "x" !
Country: "x"
Using Alsa sink with format: S16
Loading <x> with Spotify URI <spotify:track:x>
<x> (190000 ms) loaded
```

</details>

**Compilation flags** 
- [ ] dbus_mpris
- [ ] dbus_keyring
- [ ] alsa_backend
- [ ] portaudio_backend
- [x] pulseaudio_backend
- [ ] rodio_backend

**Versions (please complete the following information):**

- OS: Raspbian GNU/Linux 11 \n \l
- Spotifyd: v0.3.3

**Comment by rwjack** [comment]
Sucks I had to write all this up just to fix it myself :shrug: 

TLDR;
Run systemd services as pi instead of root (--user flag). Killed 2 flies with one swing (increased security + it actually works).

Note you can't ssh as root and `su - pi`, you have to ssh in as pi.


`pulseaudio.service`:
```
[Unit]
Description=Pulseaudio Sound Service
Requires=pulseaudio.socket

[Service]
Type=notify
ExecStart=/usr/bin/pulseaudio --verbose --daemonize=no
#ExecStartPost=/usr/bin/pactl load-module module-alsa-sink
Restart=on-failure

[Install]
Also=pulseaudio.socket
WantedBy=default.target
```

`pulseaudio.socket`:
```
[Unit]
Description=Pulseaudio Sound System

[Socket]
Priority=6
Backlog=5
ListenStream=%t/pulse/native

[Install]
WantedBy=sockets.target
```

`spotifyd.service`:
```
[Unit]
Description=A spotify playing daemon
Documentation=https://github.com/Spotifyd/spotifyd
Wants=sound.target
After=sound.target
Wants=network-online.target
After=network-online.target

[Service]
ExecStart=/usr/local/bin/spotifyd --no-daemon
Restart=always
RestartSec=12

[Install]
WantedBy=default.target
```

And finally:
```
systemctl --user --now enable pulseaudio.service pulseaudio.socket spotifyd.service
bluetoothctl conect <device_id>
# enjoy spotifyd :)
```
